example.js

```diff
@@ -71,6 +71,21 @@ const { prompt } = require('./');
                 { title: 'Gibson' },
                 { title: 'Grant' },
             ]
+        },
+        {
+            type: 'number',
+            name: 'prompt',
+            message: 'This will be overridden',
+            onPrompt() {
+                this.no = (this.no || 0) + 1;
+                this.msg = `Enter a number (e.g. ${this.no})`;
+                new Promise(resolve => setTimeout(resolve, 1000)).then(() => {
+                    if (!this.closed) {
+                        this.onPrompt();
```

Comment:
Do you manually have to call `this.onPrompt()` inside the `onPrompt` callback?

Reply:
It's just to change the message every second but I think I could have set an interval instead.

Reply:
I wonder if it would be cleaner if the onPrompt had a `render()` callback provided as its argument in case you want to rerender inside the callback.  What do you think?